core/storage/translation/gae_models.py

```diff
@@ -34,6 +35,160 @@
 
 
+# TODO(#14537): This is a duplicate of the same TypedDict in translation_domain,
+# it should be removed after we refactor our importing.
+class TranslatedContentDict(TypedDict):
+    """Dictionary representing TranslatedContent object."""
+
+    content: str|List[str]
+    needs_update: bool
+
+
+class EntityTranslationsModel(base_models.BaseModel):
+    """Model for storing entity translations ."""
+
+    # The id of the corresponding entity.
+    entity_id = datastore_services.StringProperty(required=True, indexed=True)
+    # The type of the corresponding entity.
+    entity_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the corresponding entity.
+    entity_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The ISO 639-1 code for the language an entity is written in.
+    language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The translated content.
+    translations = datastore_services.JsonProperty(required=True)
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model does not contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'entity_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_type': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'entity_version': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'language_code': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'translations': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @staticmethod
+    def _generate_id(
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        language_code: str
+    ) -> str:
+        """The method use to generate unique id for entity translations model.
+
+        Args:
+            entity_type: str. Type of an entity.
+            entity_id: str. Id of an entity.
+            entity_version: int. Version of an entity.
+            language_code: str. Language code for a given entity.
+
+        Returns:
+            str. Returns a unique id of the form
+            [entity_type]-[entity_id]-[entity_version]-[language_code].
+        """
+        return '%s-%s-%s-%s' % (
+            entity_type, entity_id, entity_version, language_code)
+
+    @classmethod
+    def get_model(
+        cls,
+        entity_type: str,
```

Comment:
Should probably be an enum, rather than an arbitrary str?

Reply:
Updated.